Update Usage Documentation (#857)
liqotech/liqo: docs/pages/usage/peering.md

---
title: Peer with a cluster
weight: 1
---

## Overview

In Liqo, peering establishes an administrative connection between two clusters and enables the resource sharing across them.
It is worth noticing that peering is unidirectional. 

> This implies that resources can be shared only from a cluster to another and not the vice-versa. Obviously, it can be optionally be enabled bi-directionally, enabling a two-way resource sharing.

Reviewer: ```suggestion
This implies that resources can be shared only from a cluster to another and not the vice-versa.
Obviously, it can be optionally be enabled bi-directionally, enabling a two-way resource sharing.
```

### Peer with a new cluster

To peer with a cluster, you should provide some information to Liqo in order to reach that cluster.
*liqoctl* can help you by generating the command to peer with a specific cluster.

> The installation of *liqoctl* is quick and you can find the complete instructions in a [dedicated section](/installation#liqoctl)

Reviewer: ```suggestion
The installation of *liqoctl* is very simple; complete instructions are available in a [dedicated section](/installation#liqoctl).
```

> Let's suppose that you want to peer from cluster A to cluster B.

Reviewer: ```suggestion
Let's suppose that you are the Cluster A and you would like to peer with Cluster B.
```

> This means that we would like to offload pods in A to the cluster B.

Reviewer: ```suggestion
This means that you would like to offload Cluster A's pods in Cluster B.
```

#### Generate the peer command

> First, pointing cluster B:

Reviewer: I did not understand the meaning of "pointing cluster B". Are you still in cluster A?

```bash
export KUBECONFIG=kubeconfig-cluster-b
liqoctl generate-add-command
```

You will obtain an output like the following:

```bash
liqoctl add cluster clusterB --auth-url https://172.18.0.5:32714 \ 
    --id 3623b0bd-3c32-4dec-994b-fc80d9d0d91d \
    --token b13b6932ee6fd890a1abe212dc21253aa6d74565fead54
```

#### Peer with a cluster

> Than, to peer A with a B cluster, you can just run the command you obtained by the previous step, after having exported the cluster A KUBECONFIG:

Reviewer: ```suggestion
Now, to peer Cluster A with B, you can just run the command you obtained by the previous step, after having exported the KUBECONFIG of Cluster A:
```

```bash
export KUBECONFIG=kubeconfig-cluster-a
liqoctl add cluster clusterB --auth-url https://172.18.0.5:32714 \ 

>     --id 3623b0bd-3c32-4dec-994b-fc80d9d0d91d \

Reviewer 2: ```suggestion
    --id 28738063-ca76-49b6-86c6-12157f21d6db \
```

    --token b13b6932ee6fd890a1abe212dc21253aa6d74565fead54
```

> If this command is executed successfully, you have completed a peering. 

Reviewer: Say how can we check that the command completed successfully.

> ### Check the status of the peering

Reviewer: ```suggestion
```

> To check the result of this addition, you can observe if new foreign clusters are available.

Reviewer: ```suggestion
To check if the above command completed successfully, you can observe if a new foreign cluster B is available.
```

```bash
kubectl get nodes
```

You should see something like:

```bash
liqo-3623b0bd-3c32-4dec-994b-fc80d9d0d91d   Ready    agent    1m   v1.19.11   alpha.service-controller.kubernetes.io/exclude-balancer=true,beta.kubernetes.io/os=linux,kubernetes.io/hostname=liqo-3623b0bd-3c32-4dec-994b-fc80d9d0d91d,kubernetes.io/role=agent,liqo.io/type=virtual-node,node.kubernetes.io/exclude-from-external-load-balancers=true,type=virtual-kubelet,liqo.io/provider=kubeadm
liqo-cluster-2-control-plane                Ready    master   87m   v1.19.11   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=liqo-cluster-2-control-plane,kubernetes.io/os=linux,node-role.kubernetes.io/master=
liqo-cluster-2-worker                       Ready    <none>   86m   v1.19.11   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=liqo-cluster-2-worker,kubernetes.io/os=linux
liqo-cluster-2-worker2                      Ready    <none>   86m   v1.19.11   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=liqo-cluster-2-worker2,kubernetes.io/os=linux
```

> The first node of the list is a virtual node pointing to the cluster B you just peered.

Reviewer: ```suggestion
The first node of the list is a virtual node pointing to the cluster B you have just peered with.
```

### What's next?

> * If you are not familiar with Liqo pod offloading, you can check out our tutorials in [Getting Started section](/gettingstarted).

Reviewer: ```suggestion
* If you are not familiar with Liqo pod offloading, you can check our tutorials in the [Getting Started](/gettingstarted) section.
```

> * If you are familiar with Liqo offloading, you may want to get in touch with advanced features explained in [namespace offloading](/namespace_offloading) section.

Reviewer: ```suggestion
* If you are already familiar with Liqo offloading, you may want to explore the advanced features presented in the [namespace offloading](/namespace_offloading) section.
```
